src/main/java/filter/Parser.java

```diff
@@ -126,6 +127,8 @@ private FilterExpression parseKeyword(Token token) {
 
     private int getInfixPrecedence(Token token) {
         switch (token.getType()) {
+        case SEMICOLON:
+            return Precedence.SHORTER_DISJUNCTION;
```

Comment:
(This is a response to another comment?)

But okay, I think it may not be worth the trouble. We can leave it as a parse error and just take `sort` to be part of the syntax.